### PR TITLE
Add support for ImagingProtocol assay_type (#77)

### PR DESCRIPTION
**Pull request by danielsotirhos**

### CODE REVIEW

File: src/humancellatlas/data/metadata/api.py [hunk 1]
```diff
@@ -450,6 +450,16 @@ def is_sequencing_process(self):
         return True
 
 
+@dataclass(frozen=True)
+class ImagingTarget:
```
Comment (danielsotirhos) [inline src/humancellatlas/data/metadata/api.py]: Thoughts on the class name? Technically it should be `Target` but it seemed a little too vague.

Comment (hannes-ucsc) [reply src/humancellatlas/data/metadata/api.py]: SGTM

File: src/humancellatlas/data/metadata/api.py [hunk 2]
```diff
@@ -535,7 +545,12 @@ class IpscInductionProtocol(Protocol):
 
 @dataclass(init=False)
 class ImagingProtocol(Protocol):
-    pass
+    target: List[ImagingTarget]
```
Comment (hannes-ucsc) [inline src/humancellatlas/data/metadata/api.py]: Didn't we agree on `Set[…]` due to precedent in `Project`?

File: test/test.py [hunk 3]
```diff
@@ -517,6 +518,17 @@ def test_analysis_protocol(self):
...
         self.assertEqual(analysis_protocols[0].protocol_name, None)
 
+    def test_imaging_protocol(self):
+        uuid = '94f2ba52-30c8-4de0-a78e-f95a3f8deb9c'
+        version = '2019-04-03T103426.471000Z'
+        manifest, metadata_files = self._load_bundle(uuid, version, replica='aws', deployment='staging')
+        bundle = Bundle(uuid, version, manifest, metadata_files)
+        imaging_protocols = [p for p in bundle.protocols.values() if isinstance(p, ImagingProtocol)]
+        self.assertEqual(len(imaging_protocols), 1)
+        self.assertEqual(len(imaging_protocols[0].target), 240)
```
Comment (hannes-ucsc) [inline test/test.py]: first() or one() from more_itertools instead of `[0]`?

Comment (danielsotirhos) [reply test/test.py]: more_itertools is not currently imported into the metadata-api project. If it is ok to add the import of the module I will do so.

File: src/humancellatlas/data/metadata/api.py (hunk 2 above)
Comment (danielsotirhos) [reply src/humancellatlas/data/metadata/api.py]: Using a set would collapse all the `ImagingTarget` objects with the same `assay_type` value into 1 object which would work negate our ability to use a `FrequencySetAccumulator` in Azul.

File: test/test.py (hunk 3 above)
Comment (hannes-ucsc) [reply test/test.py]: I wouldn't want it as a main dependency, but as a test dependency it should be ok.

File: src/humancellatlas/data/metadata/api.py (hunk 2 above)
Comment (hannes-ucsc) [reply src/humancellatlas/data/metadata/api.py]: List is fine, just briefly explain why in comment above.

**Pull request merged**